### ISSUE TITLE
If the Android client is suddenly stopped?

### ISSUE THREAD

**Issue by theprogrammer67** [Genymobile/gnirehtet#64]
If the Android client is suddenly stopped, the server does not know about it?
I made the client start on the server on the timer every minute.
This is not superfluous?

**Comment by rom1v** [comment]
> If the Android client is suddenly stopped, the server does not know about it?

Yes, it detects the disconnection. It should print something like:

```
INFO TunnelServer: Client #0 disconnected
```

> I made the client start on the server on the timer every minute.

You mean executing `gnirehtet start`? This is superfluous if the `adb reverse` tunnel is still alive (i.e. typically the device is not unplugged).

Otherwise, see https://github.com/Genymobile/gnirehtet/issues/47 and https://github.com/Genymobile/gnirehtet/issues/49.

**Comment by theprogrammer67** [comment]
>  This is superfluous if the adb reverse tunnel is still alive (i.e. typically the device is not unplugged).
But if the service on the client stopped due to an error? Everything can be)
Or is the probability very small?

**Comment by rom1v** [comment]
> But if the service on the client stopped due to an error?

If this happens, this is a bug.

Is it possible? Yes, bugs are possible.

**Comment by theprogrammer67** [comment]
>  Otherwise, see #47 and #49.

That's exactly what I did:
```
procedure TRevTethering.DoUSBArrival(Sender: TObject);
begin
  if not FRunning then
    Exit;

  TLog.WriteToLog(Self.ClassName, 'USB arrivaal');
  if UpdateDevCount(2000) then
    StartClient;
end;
```

**Comment by theprogrammer67** [comment]
thank you, Roman!

**Comment by rom1v** [comment]
> That's exactly what I did

OK :)

Then I close this issue.
